src/dstore/pmix_esh.c

```diff
@@ -228,22 +491,64 @@ int _esh_init(pmix_info_t info[], size_t ninfo)
         return PMIX_SUCCESS;
     }
     else {
+        int rc = 0;
+        int tbl_idx;
+        nspace_tbl_entry_t *nspace_tbl_entry = NULL;
+
+        if (NULL != info) {
+            for (n=0; n < ninfo; n++) {
```

Comment:
Actually I don't think that clients need to deal with jobuid at all. All you need is to do `getuid` and use it everywhere. Application process are not supposed to deal with jobuid's.